docs/source/buffy.rst

```diff
@@ -1,5 +1,143 @@
+
+• :ref:`Histogram creation <Histogram>`
+• :ref:`Signal Search <Signal-Search>`
+• :ref:`Synchronisation and trigger separation <Trigg-Separation>`
+
+.. note::
+  Historically, Buffy is based in an initial code of detsim (https://github.com/next-exp/IC/tree/master/invisible_cities/detsim) and most of its functions are located in that path but they are independent to :doc:`detsim` city.
+
+
+.. _Histogram:
+
+Histogram creation
+::::::::::::::::::
+
+As it was highlighted earlier, NEXUS information about sensor hits (``/MC/sns_response``) comes binned in time based on when a sensor sees some energy deposition.
+This means that ``time_bin`` column numbers are increasing for a given event, but they can have gaps, since the time bins where no charge is detected
+do not appear in the table. This initial part of the city, checks the time stamp of an event according to the sensors response, and defines histograms of charge distribution
+between [:math:`t_{min}`, :math:`t_{max}`], being:
+
+• :math:`t_{min}`: the time stamp of the first charge deposition of the event,
+• :math:`t_{max}`: defined considering that ``max_time`` =  :math:`t_{max}` - :math:`t_{min}`.
+
+.. image:: images/buffy/histogram_creation.png
+  :width: 800
+
+In this step, it also effectively pads with zeros in between separate signals. When these histograms (one for PMTs and another for SiPMs) are defined, they are sampled according to the binning of each sensor (``pmt_width`` and ``sipm_width``).
+Sampling widths are included in the simulation parameters (``/MC/info``), and depends on the type of sensor and detector. Normally corresponds to 25 :math:`ns` for PMTs and and 1 :math:`\mu s`\ s for SiPMs.
+
+.. _Signal-Search:
+
+Signal Search
+::::::::::::::::::
+
+Once charge is distributed in previously defined histograms, the code searches for signal-like events.
+It takes PMT sum histogram and looks for the first value of the binned charge above a certain threshold (``trigger_threshold``), and defines the trigger time, :math:`t_{trigger}`.
+Waveforms are therefore defined for PMTs:
+
+• shifting the times of the charge histogram such that the first value over threshold (:math:`t_{trigger}`) falls at the time defined as ``pre_trigger``;
+• considering a specific length based on ``buffer_length``/``pmt_width``.
+
+.. image:: images/buffy/bufferisation.png
+  :width: 800
+
+
+.. note::
+  :math:`t_{min}` does not need to be at 0, since it is defined based on the first charge deposition, independently if it is above the ``trigger_threshold`` or not.
+
+.. _Trigg-Separation:
+
+Synchronisation and trigger separation
+:::::::::::::::::::::::::::::::::::::::
+
+Since the sensor width is different for PMTs and SiPMs, it is necessary to align and synchronise the clocks between their waveforms. Waveforms are then sliced then according to binning (``pmt_width`` and ``sipm_width``), trigger time and configured pre-trigger (``pre_trigger``).
```

Comment:
Here is where I think we need to specify that we generate individual PMT and SiPM waveforms based on the parameters obtained with the summed waveforms.